src/full/Agda/TypeChecking/Rules/Record.hs

```diff
@@ -632,7 +632,7 @@ checkRecordProjections m r hasNamedCon con tel ftel fs = do
         then unless (getCohesion ai == Continuous)
                     -- Andrea TODO: properly update the context/type of the projection when we add Sharp
                     __IMPOSSIBLE__
-        else genericError $ "Cannot have record fields with modality " ++ show (getCohesion ai)
+        else typeError $ InvalidFieldModality (getCohesion ai)
```

Comment:
I always wonder whether GHC does the common subexpression analysis and introduces a `let coh = getCohesion ai` or not.